### ISSUE TITLE
--snapshot-volumes is not working on GCP

### ISSUE THREAD

**Issue by evaldasou** [heptio/ark#26]
Hey Heptio Team,

If I run this command : 

`ark backup create everything --exclude-namespaces=kube-system,heptio-ark --snapshot-volumes`

I get data on Google Cloud Storage about my cluster, but none of the persistent volumes are snapshoted.

ark-backup.json looks like this : 

```
{
  "kind": "Backup",
  "apiVersion": "ark.heptio.com/v1",
  "metadata": {
    "name": "everything",
    "namespace": "heptio-ark",
    "selfLink": "/apis/ark.heptio.com/v1/namespaces/heptio-ark/backups/everything",
    "uid": "4cf6f5d8-7d08-11e7-acb8-42010a80004a",
    "resourceVersion": "39434763",
    "creationTimestamp": "2017-08-09T13:40:26Z"
  },
  "spec": {
    "includedNamespaces": [
      "*"
    ],
    "excludedNamespaces": [
      "kube-system",
      "heptio-ark"
    ],
    "includedResources": [
      "*"
    ],
    "excludedResources": null,
    "labelSelector": null,
    **"snapshotVolumes": true,**
    "ttl": "24h0m0s"
  },
  "status": {
    "version": 1,
    "expiration": "2017-08-10T13:40:26Z",
    **"phase": "Completed",
    "volumeBackups": null,
    "validationErrors": null**
  }
}
```

Anything I am missing?

**Comment by ncdc** [comment]
@evaldasou could you please provide the logs from the ark server pod, at least the part around backing up persistent volumes? `kubectl logs -n heptio-ark deployment/ark`

**Comment by evaldasou** [comment]
hey @ncdc , sometimes when You ask You know what's the issue and the answer :)
I was using incorrect zone for the snapshots : 
`-abdb-42010a800083", volume-id "gke-cluster-1-14995ce5-pvc-ff5efe2d-71d9-11e7-abdb-42010a800083": googleapi: Error 400: Invalid value 'US-CENTRAL1'. Values must match the following regular expression: '[a-z](?:[-a-z0-9]{0,61}[a-z0-9])?', invalidParameter`

```
persistentVolumeProvider:
  gcp:
    project: my-project
    zone: us-central1-f
```
Thanks and apologize for disruption , my fault :)

**Comment by ncdc** [comment]
No worries, we're glad that you've been trying it out. Please keep the feedback coming 😄

**Comment by djschny** [comment]
In this scenario should the credentials/zone be verified before starting the backup and appropriate response returned to the user from the CLI?

**Comment by ncdc** [comment]
> In this scenario should the credentials/zone be verified before starting the backup and appropriate response returned to the user from the CLI?

@skriss can you take a look and see how much we can reasonably verify up front? We can file a separate issue for this.

**Comment by evaldasou** [comment]
@djschny GCP config looks like this : 

```
persistentVolumeProvider:
  gcp:
    project: my-project
    zone: us-central1-f
backupStorageProvider:
  bucket: my-kubernetes-backups
  gcp:
    project: my-project
    zone: US-CENTRAL1
```
Backup and snapshots has different zones. Backup works itself, but no error about snapshots visible for user directly in CLI.

**Comment by ncdc** [comment]
This is a server configuration issue... What would you like to see happen in this instance:

1. Detect issue as soon as ark-server starts, log issue, shut down ark-server (will end up in a crash loop via the kubelet)
1. Detect issue when backup/restore created, mark backup/restore as FailedValidation, include reason in backup/restore status
1. ???

**Comment by djschny** [comment]
I'm just trying to think from the perspective of how would I detect this? Right now I would probably take the approach of setting up a monitor that views periodically views the ark pod logs and looks for that particular error string and then fire an email/page if I see it.

Are there log levels in the logs [INFO, WARN, FATAL, ERROR, etc] so that one could look for a ERROR/FATAL instead of looking for a specific error message text?

**Comment by ncdc** [comment]
We have the ability to log at info, warning, and error levels.

**Comment by evaldasou** [comment]
guys , maybe not the best place to ask but how to delete backups ?
I can delete backups from my object storage, but it does not synchronize with ark client.

**ark backup get**: 

```
NAME              STATUS      CREATED                         EXPIRES   SELECTOR
nginx-backup      Completed   2017-08-09 12:33:42 +0000 UTC   22h       app=nginx
nginx-backup-1    Completed   2017-08-09 12:55:40 +0000 UTC   22h       app=nginx
nginx-backup-2    Completed   2017-08-09 13:15:13 +0000 UTC   23h       app=nginx
nginx-backup-3    Completed   2017-08-09 13:19:20 +0000 UTC   23h       app=nginx
nginx-backup-4    Completed   2017-08-09 13:19:51 +0000 UTC   23h       app=nginx
nginx-backup-5    Completed   2017-08-09 13:54:52 +0000 UTC   23h       app=nginx
```
How to delete these above from my client?

**Comment by ncdc** [comment]
@evaldasou they'll be deleted automatically when they expire. Alternatively, once you've removed them from object storage, you can `kubectl -n heptio-ark delete backup/<backup name>`.

We know this is an area we need to clean up, and we haven't yet figured out the best UX here.

**Comment by ncdc** [comment]
Please feel free to post to our Google Group (https://groups.google.com/forum/#!forum/heptio-ark) or create separate issues.

**Comment by evaldasou** [comment]
thanks will do !
that's really helpful! @ncdc !

**Comment by skriss** [comment]
@ncdc yes - I'll take a look at server config validation
